tradeexecutor/analysis/trade_analyser.py

```diff
@@ -457,8 +482,29 @@ def get_open_positions(self) -> Iterable[Tuple[PrimaryKey, TradePosition]]:
                 if position.is_open():
                     yield pair_id, position
 
-    def calculate_summary_statistics(self, time_bucket: Optional[TimeBucket] = None) -> TradeSummary:
-        """Calculate some statistics how our trades went."""
+    def calculate_summary_statistics(self, *, raw_timeline: Optional[pd.DataFrame] = None, stop_loss_pct: Optional[float] = None, time_bucket: Optional[TimeBucket] = None) -> TradeSummary:
+        """Calculate some statistics how our trades went.
+            raw_timeline and stop_loss_pct need only be provided if user wants complete list of summary statistics,
+            otherwise, the user will receive a shortened list of stats.
+
+            :param raw_timeline:
+            Created from the expand_timeline_raw() method, it only returns raw data instead of formatted strings
+            which allows easy statistical calculations for when summary stats depend on timeline.
+            
+            :param stop_loss_pct:
+            stop loss percentage
+
+            :param time_bucket:
+            time bucket to display average duration as 'number of bars' instead of 'number of days'. 
+        """
+
+        if(raw_timeline is not None):
+            assert isinstance(raw_timeline, pd.DataFrame), "Not a valid timeline, use expand_timeline_raw() method to create it"
+        if(stop_loss_pct is not None):
```

Comment:
Stop loss percent cannot be given as a parameter here. It is part of `TradingPosition` instance. 

- Each position can have its own stop loss. 
- Stop loss must be passed around other position parameters when the timeline is created

Reply:
According to @tsorro, the `max capital risk at stop loss` is calculated as follows:

How much of the total capital do we risk with the calculated stop loss.  

$$\frac{Current stop loss percentage * the position size}{Current total capital in the strategy} * 100$$

Reply:
You can get all this information as the part of the `Portfolio` object. 

- Total capital in the strategy: `portfolio.get_total_equity()` - https://tradingstrategy.ai/docs/programming/api/execution/help/tradeexecutor.state.portfolio.Portfolio.html#tradeexecutor.state.portfolio.Portfolio.get_total_equity
- Position size: For position we need to add a new method `position.get_max_value()` that is the maximum value of the position over its lifetime. The position value can move when additional trades are made. (ticket 1)
- - Note that the total capital of the strategy is based on the current (at the end of the backtest value) - we would need to add a new position variable `position.portfolio_value_at_open` so we can calculate the risk at the time of the opening the position (ticket 2)
- Then when the timeline data is exported - we have the access these functions and export them part of the analysis (ticket 3)